services/github/github-auth-service.js

```diff
@@ -1,18 +1,44 @@
 'use strict'
 
 const BaseJsonService = require('../base-json')
+const { staticAuthConfigured } = require('./github-helpers')
 
-module.exports = class GithubAuthService extends BaseJsonService {
```

Comment:
Create a second base class which uses github auth only when static auth is present, to support the use case described in #1721.

Reply:
This is more of a question stemming from my own ignorance around the complexity involved with the Github Services 😄 

If I was writing a new Github service (or porting one of the existing ones from the `LegacyService` model), how would I know which one of these base classes to use?